Jenkinsfile

```diff
@@ -3,45 +3,106 @@
 
 standardProperties()
 
-def runTests(String name, String bedrockCommand, Boolean runAll) {
-  // Clean out the old XML files before running tests, since we junit import *.XML files
-  dir('scratch') {
-    if (isUnix()) {
-      sh "rm -f *.xml"
-    } else {
-      bat "del *.xml"
+def withRemoteCreds(String provider, Closure body) {
+  if (provider == 'aws') {
+    withAwsCreds {
+      body()
+    }
+  }
+  if (provider == 'lambdatest') {
+    withLtCreds {
+      body()
+    }
+  }
+}
+
+def withAwsCreds(Closure body) {
+  tinyAws.withAWSWebIdentityToken {
+    tinyAws.withAWSRoleARN('arn:aws:iam::103651136441:role/ci/build/tinymce_devicefarm') {
+      body()
     }
   }
+}
 
-  def command = runAll ? bedrockCommand + ' --ignore-lerna-changed=true' : bedrockCommand
-  def testStatus = exec(script: command, returnStatus: true)
+def withLtCreds(Closure body) {
+  withCredentials([usernamePassword(credentialsId: 'lambda-engci', usernameVariable: 'LT_USERNAME', passwordVariable: 'LT_ACCESS_KEY')]) {
+    body()
+  }
+}
 
-  echo "Writing JUnit results for ${name} on node: $NODE_NAME"
+def runBedrockTest(String name, String command, Boolean runAll, int retry = 0, int timeout = 0) {
+  def bedrockCmd = command + (runAll ? " --ignore-lerna-changed=true" : "")
+  echo "Running Bedrock cmd: ${command}"
+  def testStatus = sh(script: command, returnStatus: true)
   junit allowEmptyResults: true, testResults: 'scratch/TEST-*.xml'
 
-  // If the tests failed (exit code 4) then just mark it as unstable
   if (testStatus == 4) {
-    unstable("Tests failed for ${name}")
+    unstable("Test failed")
```

Comment:
So the ${name} here doesn't work anymore or did it ever work?

Reply:
I'm happy to put that back in. I figured since the `blue` interface is the most popular you can easily see which one is failing but I have put that name back in